src/util/channel_layout.rs

```diff
@@ -72,4 +74,32 @@ impl ChannelLayout {
             ChannelLayout::from_bits_truncate(av_get_default_channel_layout(number) as c_ulonglong)
         }
     }
+
+    pub fn describe(&self) -> Result<String, Error> {
+        unsafe {
+            let mut av_channel_layout = mem::zeroed();
+            match av_channel_layout_from_mask(&mut av_channel_layout, self.bits) {
+                v if v >= 0 => {}
+                e => return Err(Error::from(e)),
+            }
+
+            let mut buf = vec![0u8; 64];
+            loop {
+                match av_channel_layout_describe(
+                    &av_channel_layout,
+                    buf.as_mut_ptr() as *mut i8,
+                    buf.len(),
+                ) {
+                    v if v as usize > buf.len() => {
+                        buf.resize(v as usize + 1, 0);
+                    }
+                    v if v >= 0 => {
+                        buf.truncate(v as usize);
+                        return Ok(String::from_utf8_unchecked(buf));
```

Comment:
are we sure this contains valid utf-8? I remember this `from_utf8_unchecked` bit me here https://docs.rs/ffmpeg-next/latest/src/ffmpeg_next/util/dictionary/immutable.rs.html#38-39, so I'm a little bit paranoid with that - I think it would also be okay to use `from_utf8` and then error out if it doesn't work?

Reply:
I agree, it should be changed to from_utf8. Instead, though, I dropped the commit entirely, since `ChannelLayout` changed in ffmpeg 7.0. I may add it back later!